src/Security/Security.php

```diff
@@ -206,15 +213,120 @@ class Security extends Controller implements TemplateGlobalProvider
      * @var boolean If set to TRUE or FALSE, {@link database_is_ready()}
      * will always return FALSE. Used for unit testing.
      */
-    static $force_database_is_ready = null;
+    protected static $force_database_is_ready = null;
 
     /**
      * When the database has once been verified as ready, it will not do the
      * checks again.
      *
      * @var bool
      */
-    static $database_is_ready = false;
+    protected static $database_is_ready = false;
+
+    /**
+     * @var array available authenticators
+     */
+    protected static $authenticators = [];
+
+    /**
+     * @return array
+     */
+    public static function getAuthenticators()
+    {
+        return self::$authenticators;
+    }
+
+    /**
+     * @param array|\SilverStripe\Security\Authenticator $authenticators
+     */
+    public static function setAuthenticators(array $authenticators)
+    {
+        self::$authenticators = $authenticators;
+    }
+
+    /**
+     * @var Member Currently logged in user (if available)
+     */
+    protected static $currentUser;
```

Comment:
Positioning is awkward. Please fix.

Please use protected static; this will otherwise be treated as config.